src/main/java/com/example/demo/repositories/OrderItemRepository.java

```diff
@@ -1,9 +1,13 @@
 package com.example.demo.repositories;
 
 
+import com.example.demo.persistence.entity.OrderItem;
+import com.example.demo.persistence.entity.Product;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
 public interface OrderItemRepository extends JpaRepository<OrderItem, Integer> {
     Optional<OrderItem> findByProduct(Product product);
```

Comment:
```suggestion
    List<OrderItem> findAllByProductId(Long productId);
```
Только чекни какой там тип айди у product. И я не помню какая у вас модель данных, если вы так сделали что одному продукту может соответствовать только один order item, то оставляй optional, если нет то так как я написал